src/main/java/frc/robot/subsystems/ClimberSubsystem.java

```diff
@@ -0,0 +1,94 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.REVLibError;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.ControlType;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkLimitSwitch;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.ClimberConstants;
+import frc.robot.utilities.SparkUtil;
+
+public class ClimberSubsystem extends SubsystemBase {
+  private final SparkFlex m_motor;
+  private final RelativeEncoder m_encoder;
+  private final SparkClosedLoopController m_PIDController;
+  private final SparkLimitSwitch m_limitSwitch;
```

Comment:
We're going to use a REV through-bore encoder for absolute position, and we won't have a limit switch. So you can rip out the limit switch stuff and add support for absolute angle provided by the encoder.